vendor/processors/stripe_processor.py

```diff
@@ -341,26 +352,46 @@ def build_price(self, offer, price):
         if 'stripe' not in offer.meta or 'product_id' not in offer.meta['stripe']:
             raise TypeError(f"Price cannot be created without a product_id on offer.meta['stripe'] field")
 
+        if isinstance(price, (int, float)):
+            currency = DEFAULT_CURRENCY
+            unit_amount = self.convert_decimal_to_integer(price)
+            is_price_obj = False
+        else:
+            currency = price.currency
+            unit_amount = self.convert_decimal_to_integer(price.cost)
+            is_price_obj = True
```

Comment:
This else will never happen in the sync if you always call offer.current_price()